From a9a9fd9e822c9eb1c2fe7a2b3e97e0660052c0c9 Mon Sep 17 00:00:00 2001
From: Amos Lim <eui-sang.lim@samsung.com>
Date: Mon, 25 Feb 2019 04:19:10 +0000
Subject: [PATCH] service worker: Remove Cookie handling from the test helper

Move On*Event(Stub) functions from EmbeddedWorkerTestHelper to
FakeServiceWorker.

Bug: 934665
Change-Id: I7914d98a3d0dc4ac67edc9dc9b4a02111c73c580
Reviewed-on: https://chromium-review.googlesource.com/c/1484966
Commit-Queue: Matt Falkenhagen <falken@chromium.org>
Reviewed-by: Matt Falkenhagen <falken@chromium.org>
Cr-Commit-Position: refs/heads/master@{#635030}
---
 .../embedded_worker_test_helper.cc              | 17 -----------------
 .../embedded_worker_test_helper.h               | 10 ----------
 .../service_worker/fake_service_worker.cc       |  2 +-
 3 files changed, 1 insertion(+), 28 deletions(-)

diff --git a/content/browser/service_worker/embedded_worker_test_helper.cc b/content/browser/service_worker/embedded_worker_test_helper.cc
index 5b13ef91dc5b0..2863309dacf7f 100644
--- a/content/browser/service_worker/embedded_worker_test_helper.cc
+++ b/content/browser/service_worker/embedded_worker_test_helper.cc
@@ -374,13 +374,6 @@ void EmbeddedWorkerTestHelper::OnBackgroundFetchSuccessEvent(
   std::move(callback).Run(blink::mojom::ServiceWorkerEventStatus::COMPLETED);
 }
 
-void EmbeddedWorkerTestHelper::OnCookieChangeEvent(
-    const net::CanonicalCookie& cookie,
-    ::network::mojom::CookieChangeCause cause,
-    blink::mojom::ServiceWorker::DispatchCookieChangeEventCallback callback) {
-  std::move(callback).Run(blink::mojom::ServiceWorkerEventStatus::COMPLETED);
-}
-
 void EmbeddedWorkerTestHelper::OnFetchEvent(
     int /* embedded_worker_id */,
     blink::mojom::FetchAPIRequestPtr /* request */,
@@ -505,16 +498,6 @@ void EmbeddedWorkerTestHelper::OnBackgroundFetchSuccessEventStub(
                      std::move(callback)));
 }
 
-void EmbeddedWorkerTestHelper::OnCookieChangeEventStub(
-    const net::CanonicalCookie& cookie,
-    ::network::mojom::CookieChangeCause cause,
-    blink::mojom::ServiceWorker::DispatchCookieChangeEventCallback callback) {
-  base::ThreadTaskRunnerHandle::Get()->PostTask(
-      FROM_HERE,
-      base::BindOnce(&EmbeddedWorkerTestHelper::OnCookieChangeEvent,
-                     AsWeakPtr(), cookie, cause, std::move(callback)));
-}
-
 void EmbeddedWorkerTestHelper::OnFetchEventStub(
     int embedded_worker_id,
     blink::mojom::FetchAPIRequestPtr request,
diff --git a/content/browser/service_worker/embedded_worker_test_helper.h b/content/browser/service_worker/embedded_worker_test_helper.h
index de50556165648..e9e7de1ff3485 100644
--- a/content/browser/service_worker/embedded_worker_test_helper.h
+++ b/content/browser/service_worker/embedded_worker_test_helper.h
@@ -25,9 +25,7 @@
 #include "content/browser/service_worker/service_worker_test_utils.h"
 #include "content/browser/url_loader_factory_getter.h"
 #include "mojo/public/cpp/bindings/binding.h"
-#include "net/cookies/cookie_change_dispatcher.h"
 #include "net/http/http_response_info.h"
-#include "services/network/public/mojom/cookie_manager.mojom.h"
 #include "third_party/blink/public/common/service_worker/service_worker_status_code.h"
 #include "third_party/blink/public/mojom/service_worker/embedded_worker.mojom.h"
 #include "third_party/blink/public/mojom/service_worker/service_worker.mojom.h"
@@ -188,10 +186,6 @@ class EmbeddedWorkerTestHelper {
       blink::mojom::BackgroundFetchRegistrationPtr registration,
       blink::mojom::ServiceWorker::DispatchBackgroundFetchSuccessEventCallback
           callback);
-  virtual void OnCookieChangeEvent(
-      const net::CanonicalCookie& cookie,
-      ::network::mojom::CookieChangeCause cause,
-      blink::mojom::ServiceWorker::DispatchCookieChangeEventCallback callback);
   virtual void OnFetchEvent(
       int embedded_worker_id,
       blink::mojom::FetchAPIRequestPtr request,
@@ -268,10 +262,6 @@ class EmbeddedWorkerTestHelper {
       blink::mojom::BackgroundFetchRegistrationPtr registration,
       blink::mojom::ServiceWorker::DispatchBackgroundFetchSuccessEventCallback
           callback);
-  void OnCookieChangeEventStub(
-      const net::CanonicalCookie& cookie,
-      ::network::mojom::CookieChangeCause cause,
-      blink::mojom::ServiceWorker::DispatchCookieChangeEventCallback callback);
   void OnExtendableMessageEventStub(
       blink::mojom::ExtendableMessageEventPtr event,
       blink::mojom::ServiceWorker::DispatchExtendableMessageEventCallback
diff --git a/content/browser/service_worker/fake_service_worker.cc b/content/browser/service_worker/fake_service_worker.cc
index 77f09a5617c7f..8250d1e7226b9 100644
--- a/content/browser/service_worker/fake_service_worker.cc
+++ b/content/browser/service_worker/fake_service_worker.cc
@@ -101,7 +101,7 @@ void FakeServiceWorker::DispatchCookieChangeEvent(
     const net::CanonicalCookie& cookie,
     ::network::mojom::CookieChangeCause cause,
     DispatchCookieChangeEventCallback callback) {
-  helper_->OnCookieChangeEventStub(cookie, cause, std::move(callback));
+  std::move(callback).Run(blink::mojom::ServiceWorkerEventStatus::COMPLETED);
 }
 
 void FakeServiceWorker::DispatchFetchEvent(
